### DISCUSSION TITLE
Differentiating built-in Vs user clients

### DISCUSSION THREAD

**Discussion by ossdhaval** [JanssenProject/jans#10238] [Ideas]
On a Janssen server with all modules installed, there are number of clients created to be used by various modules. See below:

![image](https://github.com/user-attachments/assets/4aa175e5-fe7d-411e-a8dd-ea29695a9173)

These built-in clients have nothing to differentiate them from a user created client. This will make it difficult for user's to find their own client from the built-in ones.

I can think of two ways to make it better:

1. Put these built-in clients in a separate section to visually differentiate them. These clients will be rarely touched by the admins anyway.
2. Follow a naming convention for naming all the preinstalled clients like starting with `jans`. This will give the Admin some indication that these are preinstalled ones.

Does everyone see this as a problem? More suggestions are welcome.

**Comment by pujavs** [comment]
For config-api, the id series is unique for app client, test-client and the config-api related oauth2 scope, all have the same series that is `180..`

**Reply by ossdhaval** [reply]
True, that applies only to the config API.  But there are many other clients too as shown in the screenshot. There should be something that clearly tells the difference.

**Reply by devrimyatar** [reply]
> True, that applies only to the config API. But there are many other clients too as shown in the screenshot. There should be something that clearly tells the difference.

The same is true for all clients created during setup

**Comment by yuriyz** [comment]
I vote to prefix name with "Built-in" word so it's clear at glance that client is built-in.

**Comment by nynymike** [comment]
They are created during setup, not sure they are inherently "built-in".

**Reply by yuriyz** [reply]
True, it's not. I guess we need some better word for it.

**Comment by devrimyatar** [comment]
@ossdhaval @nynymike @yuriyz There is a differention. Builtin clients are in this format
`SSNN.uuid`
Where `SS` is service number, and `NN` is client number. `uuid` is Universally Unique Identifier.
For example saml clients starts with **21**, casa clients starts with **30**. We discussed this with @yurem when I first faced difficulty in upgrade process for Gluu CE, several years ago.

**Reply by nynymike** [reply]
Maybe we can add some extra warning for the admin. But these are still clients... And if the admin wants to delete them, they must have their reasons (and accept the consequences)

**Reply by yuriyz** [reply]
Should we maybe prefix client's with "Setup created"? We need somehow indicate at glance that client is pre-created by setup. Also on deletion we clearly should state that admin can go on and delete it but it may break some functionality.

**Reply by ossdhaval** [reply]
@yuriyz has echoed my concern. `At a glance` is the key here. The differentiator should be explicit and need not be derived.

**Reply by ossdhaval** [reply]
**Option-1**

Prefix with `setup created`

i.e `SCIM client`  becomes `setup created SCIM client`

**Option-2**

How about prefixing with `system` and user hyphen instead of space

i.e `SCIM client` becomes `system-scim-client`

cc: @yuriyz @devrimyatar @nynymike

**Reply by yuriyz** [reply]
Looks good. And yes, we do need delimiter in between. E.g. `Setup created - SCIM Client` or `System - Scim Client`. Both looks good to me.
